packages/components/src/slot-fill/bubbles-virtually/fill.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, useEffect, createPortal } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useSlot from './use-slot';
+
+export default function Fill( { name, children } ) {
```

Comment:
This is a simple and nice implementation but I believe we had issues with fills being reordered when we switch between blocks (toolbar and inspector) and we had to add a custom logic to keep the fills rendering in the exact same order they appear on the React tree (even with conditions).

I see that this new implementation passes the existing unit tests right which I believe account for this (and I'm surprised it does without any tweaks to the rendering order). We might just want to confirm in the UI too.

Reply:
Hmm, from what I can see, the current unit tests are not covering that case for `bubblesVirtually`. Actually, it seems that `bubblesVirtually` is not being covered at all? `react-test-renderer` doesn't seem to support portals. But I'll check if `react-dom/test-utils` can do the job.

Also, I can't find how it's been solved for the current implementation (using `bubblesVirtually`). Do you have an idea?

I'm pretty sure that my implementation doesn't cover that. `ReactDOM.createPortal`, by default, will just attach elements in the order they're rendered.

I've just made some experiments with the `key` prop here: https://codesandbox.io/s/keeping-react-portal-order-uk9xb

Reply:
It's related to the `occurrence` counter we have on fills and there's  a function to reorder the fills in the Provider I believe. That said, If we manage to run the unit tests on the new implementation, that would be enough to validate it I think.